server/src/main/java/com/thoughtworks/go/config/update/RoleConfigCommand.java

```diff
@@ -60,12 +64,12 @@ public boolean canContinue(CruiseConfig cruiseConfig) {
     public boolean isValid(CruiseConfig preprocessedConfig) {
         preprocessedRole = preprocessedConfig.server().security().getRoles().findByNameAndType(role.getName(), role.getClass());
 
-        if (!preprocessedRole.validateTree(RolesConfigUpdateValidator.validationContextWithSecurityConfig(preprocessedConfig))) {
-            BasicCruiseConfig.copyErrors(preprocessedRole, role);
-            return false;
-        }
+        preprocessedRole.validateTree(RolesConfigUpdateValidator.validationContextWithSecurityConfig(preprocessedConfig));
+
+        List<ConfigErrors> allErrors = ErrorCollector.getAllErrors(preprocessedRole);
+        BasicCruiseConfig.copyErrors(preprocessedRole, role);
 
-        return true;
+        return allErrors.isEmpty();
```

Comment:
`BasicCruiseConfig.copyErrors` is an iterator over config object. Can we avoid traversing the entire config object (including children) if there are no errors by changing this to:
```java
        List<ConfigErrors> allErrors = ErrorCollector.getAllErrors(preprocessedRole);
        boolean isEmpty = allErrors.isEmpty();
        if(!isEmpty) {
                BasicCruiseConfig.copyErrors(preprocessedRole, role);
        }
        return isEmpty;
```